File: frontend/src/types/pharmacogenomics.ts
export interface PgxVariant {
  variant_id: string
  chrom: string
  pos: number
  ref: string
  alt: string
  rs_id: string | null
  gene: string
  allele: string
  function: string
  change: string
  gene_symbol: string | null
  consequence: string | null
}

export interface DrugInteraction {
  drug: string
  category: string
  interaction: string
  impact: string
  metabolizer_status: string
  guideline: string
}

export interface AlleleInfo {
  allele: string
  function: string
  rs_id: string | null
}

export interface GeneSummary {
  metabolizer_status: string
  alleles: AlleleInfo[]
  drug_count: number
}

export interface AlleleReference {
  rsid: string | null
  change: string
  function: string
}

export interface GeneReference {
  gene: string
  chromosome: string | null
  alleles: Record<string, AlleleReference>
  drug_count: number
}

export interface PharmacogenomicsData {
  pgx_variants: PgxVariant[]
  drug_interactions: DrugInteraction[]
  gene_summaries: Record<string, GeneSummary>
  summary: string
  allele_reference: GeneReference[]
}

File: backend/app/schemas/pharmacogenomics.py
"""Pydantic response models for pharmacogenomics endpoint."""
from pydantic import BaseModel, Field


class PgxVariant(BaseModel):
    variant_id: str
    chrom: str
    pos: int
    ref: str
    alt: str
    rs_id: str | None = None
    gene: str
    allele: str
    function: str
    change: str
    gene_symbol: str | None = None
    consequence: str | None = None


class DrugInteraction(BaseModel):
    drug: str
    category: str
    interaction: str
    impact: str
    metabolizer_status: str
    guideline: str


class AlleleInfo(BaseModel):
    allele: str
    function: str
    rs_id: str | None = None


class GeneSummary(BaseModel):
    metabolizer_status: str
    alleles: list[AlleleInfo]
    drug_count: int


class AlleleReference(BaseModel):
    rsid: str | None = None
    change: str
    function: str


class GeneReference(BaseModel):
    gene: str
    chromosome: str | None = None
    alleles: dict[str, AlleleReference]
    drug_count: int


class PharmacogenomicsResponse(BaseModel):
    pgx_variants: list[PgxVariant]
    drug_interactions: list[DrugInteraction]
    gene_summaries: dict[str, GeneSummary]
    summary: str
    allele_reference: list[GeneReference] = Field(default_factory=list)
